2_process/src/pair_nhd_reaches.R

```diff
@@ -33,12 +37,18 @@ pair_nhd_reaches <- function(nhd_lines,prms_line){
     filter(STREAMORDE==STREAMCALC) %>%
     filter(HYDROSEQ==max(HYDROSEQ))
   
-  # Add in special handling for PRMS subsegid "341_1"
-  # [PRMS line doesn't completely overlap NHD, leading to misleading upstream HYDROSEQ id]
+  # Add in special handling for select PRMS segments 
+  # PRMS line "341_1" doesn't completely overlap NHD, leading to misleading upstream HYDROSEQ id
   if(prms_line$subsegid=="341_1"){
     nhd_paired_up <- nhd_paired %>%
       filter(COMID == "4480903")
   }
+  # The upstream COMID associated with PRMS subsegid "2765_1" likely does not drain to the 
+  # downstream COMID, leading to infinite recursion errors if omit_divergences is TRUE
```

Comment:
I found this because the `traverse_nhd()` function threw infinite recursion errors when running segment 2765_1 if `omit_divergences = TRUE`. If we identify the upstream COMID in our usual way, that supposed upstream COMID appears to drain to a completely different river segment in NHDPlusV2 (which is why we get an error when we try to traverse the flowlines until we find the expected downstream COMID). 

![2765_example](https://user-images.githubusercontent.com/8785034/171748261-8a27e4ae-f4a1-4696-9ddd-a0309d476b1b.png)

Reply:
Okay, thanks for these explanations. I don't see a good way to generalize these findings into an automatic detection